internal/sdk/cloudian/sdk.go

```diff
@@ -230,6 +236,43 @@ func (client Client) CreateUser(ctx context.Context, user User) error {
 	return resp.Body.Close()
 }
 
+// Fetch user credentials.
+func (client Client) GetUserCredentials(ctx context.Context, user User) ([]SecurityInfo, error) {
+	url := client.baseURL + "/user/credentials/list?userId=" + user.UserID + "&groupId=" + user.GroupID
+
+	req, err := client.newRequest(ctx, url, http.MethodGet, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating credentials request: %w", err)
+	}
+
+	resp, err := client.httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("error performing credentials request: %w", err)
+	}
+
+	defer resp.Body.Close() // nolint:errcheck
+
+	switch resp.StatusCode {
+	case 200:
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, fmt.Errorf("error reading credentials response: %w", err)
+		}
+
+		var securityInfo []SecurityInfo
+		if err := json.Unmarshal(body, &securityInfo); err != nil {
+			return nil, fmt.Errorf("error parsing credentials response: %w", err)
+		}
+
+		return securityInfo, nil
+	case 204:
+		// Cloudian-API returns 204 if no security credentials found
+		return nil, ErrNotFound
+	default:
+		return nil, fmt.Errorf("error: list credentials unexpected status code: %d", resp.StatusCode)
+	}
+}
```

Comment:
No test of this function?

Reply:
None of the cloudian invoking endpoints have tests. We would have to introduce some interfaces and start mocking or stubbing I guess.
I've only tested curl against the cloudian API docs

Reply:
We should probably add some kind of Go REST module that we can easily stub out. Untested code is no good.